Core.Auditing/IfxAuditLogger.cs

```diff
@@ -1,101 +1,124 @@
-        /// <param name="auditMandatoryProperties">Mandatory properties to log</param>
-        /// <param name="auditOptionalProperties">This is optional app specific properties</param>
-        public void LogApplicationAuditEvent(ITelemetryClient telemetryClient, AuditMandatoryProperties auditMandatoryProperties, AuditOptionalProperties auditOptionalProperties = null)
+        private void LogAuditEvent(ITelemetryClient telemetryClient, OperationCategory operationCategory, OperationType operationType, string operationName, OperationResult operationResult, List<CallerIdentity> callerIdentities, List<TargetResource> targetResources)
         {
-            IfxResult ifxResult = IfxResult.Failure(IfxResultCode.UnknownError, "Unknown Error");
-            bool appLogResult = IfxAudit.LogApplicationAudit(auditMandatoryProperties, auditOptionalProperties, ifxResult);
-            bool managementLogResult = IfxAudit.LogManagementAudit(auditMandatoryProperties, auditOptionalProperties, ifxResult);
+            AuditRecord auditRecord = new AuditRecord();
+
+            auditRecord.AddOperationCategory(operationCategory);
+            auditRecord.OperationType = operationType;
+            auditRecord.OperationName = operationName;
+            auditRecord.OperationResult = operationResult;
+            //auditRecord.OperationResultDescription = Required if operation result is ever Failure; 
 
-            if ((!appLogResult || !managementLogResult) && !alreadyLoggedPerSession)
+            foreach (CallerIdentity callerIdentity in callerIdentities)
             {
-                /* In local box environment, audit logging mightn't be saved.
-                In order to locally debug your code, please download and execute RunBefore_IFxAuditWebAppsLocalTesting.cmd, 
-                this sets a couple of environment variables that are present in Web Apps and IFxAudit needs.After running that please make sure you restart
-                Visual Studio so that the changes are picked up.(See https://genevamondocs.azurewebsites.net/collect/references/ifxref/ifxAuditWebApps.html)
-                */
-                Dictionary<string, string> properties = new Dictionary<string, string>()
-                {
-                    { "IfxResultMessage", ifxResult.Message },
-                    { "IfxResultCode", ifxResult.Code.ToString() },
-                };
-                telemetryClient.TrackEvent("IfxApplicationAuditFailure", properties);
-                alreadyLoggedPerSession = true;
+                auditRecord.AddCallerIdentity(callerIdentity.type, callerIdentity.name, callerIdentity.description);
             }
-            else
+
+            foreach (TargetResource targetResource in targetResources)
             {
-                // If there is ever success for sending the logs, reset "alreadyLoggedPerSession" so that we capture transient failures more accurately.
-                alreadyLoggedPerSession = false;
```

Comment:
How does this work locally? Does it fail every single time? If so, do we want to bring "alreadyLoggedPerSession"?

Reply:
This does not fail locally (although audit logs are not readable from the console)  so I do not believe that we need this flag